libandroid-navigation-ui/src/test/java/com/mapbox/services/android/navigation/ui/v5/BaseTest.java

```diff
@@ -27,4 +42,55 @@ protected String loadJsonFixture(String filename) throws IOException {
     Scanner scanner = new Scanner(inputStream, UTF_8.name()).useDelimiter("\\A");
     return scanner.hasNext() ? scanner.next() : "";
   }
+
+  protected RouteProgress buildRouteProgress(DirectionsRoute route,
```

Comment:
I think instead of putting this in `BaseTest` we should have classes that fake data for us, like `RouteProgressFaker`, just so we don't clutter up `BaseTest` with implementation specific code

Reply:
@danesfeder 
> I extracted the building logic from `BaseTest` into classes that separate the building concerns.

How did you address @devotaaabel's comment? `buildRouteProgress` is still in `BaseTest` 🤔 Could you clarify?